lib/Backends/Interpreter/InterpreterNodes.cpp

```diff
@@ -3696,67 +3696,88 @@ void BoundInterpreterFunction::fwdBatchedReduceAddInst(
                             eBatchDims, eDestDims);
 }
 
-template <typename ElemTy>
-void BoundInterpreterFunction::fwdBatchedReduceMinInstImpl(
-    Value *batch, Value *dest, const ShapeVector &eBatchDims,
-    const ShapeVector &eDestDims, ElemTy max) {
-  static_assert(max_tensor_dimensions == 6,
-                "Loops below assume max_tensor_dimensions = 6.");
-  // Get unowned handles of the batch and dest with these new expanded dims.
-  auto eBatch = getTensor(batch)->getUnowned(eBatchDims);
-  auto eDest = getTensor(dest)->getUnowned(eDestDims);
-  auto eBatchH = eBatch.getHandle<ElemTy>();
-  auto eDestH = eDest.getHandle<ElemTy>();
-  eDestH.clear(max);
-
-  unsigned int axes[max_tensor_dimensions];
-  for (dim_t i = 0; i < max_tensor_dimensions; i++) {
-    axes[i] = (eDestDims[i] > 1);
+/// Macro to define ReduceMin/Max kernel implementation.
+#define DEFINE_REDUCEMINMAX_INST_IMPL(func, compare)                           \
+  template <typename ElemTy>                                                   \
+  void BoundInterpreterFunction::fwdBatched##func##InstImpl(                   \
+      Value *batch, Value *dest, const ShapeVector &eBatchDims,                \
+      const ShapeVector &eDestDims, ElemTy init) {                             \
+    static_assert(max_tensor_dimensions == 6,                                  \
+                  "Loops below assume max_tensor_dimensions = 6.");            \
+    /* Get unowned handles of the batch and dest with these new expanded       \
+     * dims.*/                                                                 \
+    auto eBatch = getTensor(batch)->getUnowned(eBatchDims);                    \
+    auto eDest = getTensor(dest)->getUnowned(eDestDims);                       \
+    auto eBatchH = eBatch.getHandle<ElemTy>();                                 \
+    auto eDestH = eDest.getHandle<ElemTy>();                                   \
+    eDestH.clear(init);                                                        \
+                                                                               \
+    unsigned int axes[max_tensor_dimensions];                                  \
+    for (dim_t i = 0; i < max_tensor_dimensions; i++) {                        \
+      axes[i] = (eDestDims[i] > 1);                                            \
+    }                                                                          \
+                                                                               \
+    /* We can use this loop for all shapes. Use the same indices for both      \
+     * the*/                                                                   \
+    /* batch and dest, except for setting the axis index in the dest to 0.*/   \
```

Comment:
nit: can you clean up the comment alignment here?

Reply:
Done.